plugins/window_size/gtk/window_size_plugin.cc

```diff
@@ -0,0 +1,203 @@
+
+  if (strcmp(method, kGetScreenListMethod) == 0) {
+    g_autoptr(FlValue) screens = get_screen_list(self);
+    fl_method_call_respond_success(method_call, screens, nullptr);
+  } else if (strcmp(method, kGetWindowInfoMethod) == 0) {
+    g_autoptr(FlValue) window_info = get_window_info(self);
+    fl_method_call_respond_success(method_call, window_info, nullptr);
+  } else if (strcmp(method, kSetWindowFrameMethod) == 0) {
+    if (fl_value_get_type(args) != FL_VALUE_TYPE_LIST ||
+        fl_value_get_length(args) != 4) {
+      fl_method_call_respond_error(method_call, "Bad arguments",
+                                   "Expected 4-element list", nullptr, nullptr);
+      return;
+    }
+    gint x =
+        static_cast<gint>(fl_value_get_float(fl_value_get_list_value(args, 0)));
+    gint y =
+        static_cast<gint>(fl_value_get_float(fl_value_get_list_value(args, 1)));
+    gtk_window_move(self->window, x, y);
+    gint width =
+        static_cast<gint>(fl_value_get_float(fl_value_get_list_value(args, 2)));
+    gint height =
+        static_cast<gint>(fl_value_get_float(fl_value_get_list_value(args, 3)));
+    gtk_window_resize(self->window, width, height);
+    fl_method_call_respond_success(method_call, nullptr, nullptr);
+  } else if (strcmp(method, kSetWindowTitleMethod) == 0) {
+    if (fl_value_get_type(args) != FL_VALUE_TYPE_STRING) {
+      fl_method_call_respond_error(method_call, "Bad arguments",
+                                   "Expected string", nullptr, nullptr);
+      return;
+    }
+    gtk_window_set_title(self->window, fl_value_get_string(args));
+    fl_method_call_respond_success(method_call, nullptr, nullptr);
+  } else
+    fl_method_call_respond_not_implemented(method_call, nullptr);
+}
+
+static void fl_window_size_plugin_dispose(GObject* object) {
+  FlWindowSizePlugin* self = FL_WINDOW_SIZE_PLUGIN(object);
+
+  g_clear_object(&self->channel);
+  g_clear_object(&self->window);
+
+  G_OBJECT_CLASS(fl_window_size_plugin_parent_class)->dispose(object);
+}
+
+static void fl_window_size_plugin_class_init(FlWindowSizePluginClass* klass) {
+  G_OBJECT_CLASS(klass)->dispose = fl_window_size_plugin_dispose;
+}
+
+static void fl_window_size_plugin_init(FlWindowSizePlugin* self) {}
+
+FlWindowSizePlugin* fl_window_size_plugin_new(FlPluginRegistrar* registrar) {
+  FlWindowSizePlugin* self = FL_WINDOW_SIZE_PLUGIN(
+      g_object_new(fl_window_size_plugin_get_type(), nullptr));
+
+  g_autoptr(FlStandardMethodCodec) codec = fl_standard_method_codec_new();
+  self->channel =
+      fl_method_channel_new(fl_plugin_registrar_get_messenger(registrar),
+                            kChannelName, FL_METHOD_CODEC(codec));
+  fl_method_channel_set_method_call_handler(self->channel, method_call_cb,
+                                            g_object_ref(self), g_object_unref);
+  FlView* view = fl_plugin_registrar_get_view(registrar);
+  self->window =
+      GTK_WINDOW(g_object_ref(gtk_widget_get_parent(GTK_WIDGET(view))));
```

Comment:
This pattern bit me a couple of times on macOS; it works now, but with refactoring of the view/engine interaction it may become the case later that the view is null at registration time, and set later. I ended up having to change the plugins to keep the registrar, and ask for the view/window when handling messages. We should do that up front here to avoid the same breakage I had on macOS.

Reply:
I wasn't sure if the Registrar should be long lived, but it will be a lot simpler to keep it around.